data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -185,7 +192,7 @@ private void parseCompleteStreamFromResponseHeader(final S3ObjectReference s3Obj
     private JsonNode getJsonNode(String selectObjectOptionalString) throws JsonProcessingException {
         final JsonNode recordJsonNode;
         if(serializationFormatOption.equals(S3SelectSerializationFormatOption.CSV) &&
-                FileHeaderInfo.NONE.name().equals(fileHeaderInfo.name()))
+                FileHeaderInfo.NONE.name().equalsIgnoreCase(s3SelectCSVOption.getFileHeaderInfo()))
```

Comment:
I tend to think that case should matter here. Pipelines will start to look confusing if users are using alternate cased values.

Reply:
Thanks @dlvenable for your review, we have incorporated the change without case.